src/python/pants/jvm/goals/lockfile.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from pants.core.goals.generate_lockfiles import (
+    GenerateLockfilesSubsystem,
+    Lockfile,
+    LockfileRequest,
+    WrappedLockfileRequest,
+)
+from pants.engine.fs import CreateDigest, Digest, FileContent, PathGlobs, Snapshot
+from pants.engine.rules import Get, collect_rules, rule
+from pants.engine.unions import UnionRule
+from pants.jvm.resolve import coursier_fetch, jvm_tool
+from pants.jvm.resolve.common import ArtifactRequirements
+from pants.jvm.resolve.coursier_fetch import CoursierResolvedLockfile
+from pants.jvm.resolve.jvm_tool import GatherJvmCoordinatesRequest, JvmToolBase
+from pants.jvm.resolve.key import CoursierResolveKey
+from pants.util.logging import LogLevel
+from pants.util.ordered_set import FrozenOrderedSet
+
+
+@dataclass(frozen=True)
+class JvmLockfileRequest(LockfileRequest):
+    artifact_inputs: FrozenOrderedSet[str]
+
+    @classmethod
+    def from_tool(cls, tool: JvmToolBase) -> JvmLockfileRequest:
```

Comment:
Note that before we were erroring when using this method and the DEFAULT_LOCKFILE was used. That is now instead handled by `generate_lockfiles.py`.